### ISSUE TITLE
Consume value from callback

### ISSUE THREAD

**Issue by mapogolions** [lgse/ts-results#2]
What do you think about creating a new method `TryWithData` or changing the signature of the existing `Try` method like below 

```js
public static TryWithData<T = {}>(
    callback: (...args: unknown[]) => T,
    errorTransformer?: (e: unknown) => Error | null
  ): Result<T> {
    try {
      const data = callback();
      return new Result<T>(true).WithData(data);
    } catch (e) {
      const error =
        typeof errorTransformer === 'function' ? errorTransformer(e) : null;
      return Result.Fail<T>(error);
    }
  }
```

Note that in the snippet above, the callback returns `T` instead of `any` type. As a result, we can use the outcome as data

**Comment by l0gicgate** [comment]
I think this makes a lot of sense. Modifying the existing `Try` method is probably the way to go, it was an oversight on my part. We should make the return type of callback() optional perhaps?

**Comment by mapogolions** [comment]
Let it be nullable. I've created PR #3

**Comment by l0gicgate** [comment]
Fixed by #3
